docs/src/main/tut/task.md

```diff
@@ -0,0 +1,20 @@
+---
+menu: tut
+weight: 115
+title: Task[T]
+---
+
+As you might have noticed, a few methods, e.g. `withBody` lift a `Response` into a `Task` context, so you end up with a `Task[Response]`. Coincidentially, all the methods which relate to the body of the `Request`/`Response`. Why does this happen? http4s is built on [Streaming], so directly materializing the body of a `Response` defeats any streaming benefits. Instead, the result is wrapped in a `Task` which preserves the asynchronous nature.
+
+## Methods on `Task[Response]`
+
+As with every other context, you have to map/flatMap it to do useful things with it. To avoid some of this, there's [Message], which puts quite a few methods on `Task[Request]`/`Task[Response]`. There's some more methods on [TaskRequestOps] and [TaskResponseOps].
```

Comment:
It might be good to show an example here of what things would be like without this syntax, and then with.